src/frontend/magnify/src/components/design-system/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import TextField from './TextField';
+
+describe('TextField', () => {
+  it('should follow the user typing', async () => {
+    const user = userEvent.setup();
+    const TestElement = () => {
+      const [value, setValue] = React.useState('Hello');
+      return (
+        <TextField
+          label="My input"
+          name="my-input"
+          onChange={(e) => setValue(e.target.value)}
+          value={value}
+        />
+      );
+    };
+
+    render(<TestElement />);
+
+    const input = screen.getByLabelText('My input') as HTMLInputElement;
```

Comment:
Perhaps if you get it by role you don't have to cast the type and I think it's better to do it this way.